lib/Dialect/Arc/ModelInfo.cpp

```diff
@@ -55,18 +55,26 @@ LogicalResult circt::arc::collectStates(Value storage, unsigned offset,
     if (!isa<AllocStateOp, RootInputOp, RootOutputOp, AllocMemoryOp>(op))
       continue;
 
+    SmallVector<StringAttr> names;
     auto opName = op->getAttrOfType<StringAttr>("name");
-    if (!opName || opName.getValue().empty())
+    if (opName && !opName.getValue().empty()) {
+      names.push_back(opName);
+    } else if (auto nameAttrs = op->getAttrOfType<ArrayAttr>("names")) {
+      for (auto attr : nameAttrs)
+        names.push_back(cast<StringAttr>(attr));
```

Comment:
Do we have to check that it is actually a `StringAttr` since "names" is not a verified attribute?
Also, why is it not necessary to check for empty strings in this case?